projects/components/facet/bootstrap/facet-filters/facet-filters.ts

```diff
@@ -130,4 +138,66 @@ export class BsFacetFilters implements OnChanges {
         return this.facetService.hasData(facet.aggregation, this.results);
     }
 
+    private addFacetMenu() {
+        if (!this.userFacets || this.userFacets?.length < this.facets.length) { 
+            let outFacets: Action[] = [];
+
+            outFacets.push(new Action({
+                name: `add_all`,
+                text: "All",
+                title: `Add all facets`,
+                action: () => {
+                    if (this.userSettingsService.userSettings) this.userSettingsService.userSettings["facets"] = undefined;
+                    this.userSettingsService.save();
+                    this.initUserFacets();
+                    this.buildFilters();
+                }
+            }));
+
+            for (let facet of this.facets) {
+                if (!this.userFacets?.find(userFacet => userFacet.name === facet.name)) outFacets.push(new Action({
+                    name: `add_${facet.name}`,
+                    text: facet.title,
+                    icon: facet.icon,
+                    title: `Add ${facet.title}`,
+                    action: () => {
+                        this.facetService.addFacet({name: facet.name, position: 0, hidden: false, expanded: true, view: ""});
+                        this.buildFilters();
+                    }
+                }));
+            }
+
+            let add_action = new Action({
+                name: "add_facet",
+                text: "Add",
+                icon: "fas fa-plus",
+                children: outFacets
+            });
+            this.filters = [add_action, ...this.filters]; 
+       }
+    }
+
+    get filteredFacets() {
+        let new_facets: FacetConfig[] = [];
+
+        if (this.userFacets) {
+            for (let facet of this.facets) {
+                let pos = this.userFacets.findIndex((userFacet) => userFacet.name === facet.name);
+                if (pos >= 0) new_facets.push(facet);
+            }
+        }
+        return new_facets;
+    }
+
+    private initUserFacets() {
+        for (let facet of this.facets) { 
+            if (!this.userFacets?.find(userFacet => userFacet.name === facet.name))
+                this.facetService.addFacet({name: facet.name, position: 0, hidden: false, expanded: true, view: ""});
+        }
+    }
+
+    get userFacets() {
```

Comment:
Rather than reading the user settings directly, wouldn't it be better to access this via `facetService.facets`? (it would be a bit cleaner, since you're already using other methods from the facet service=

Reply:
Didn't check the service thoroughly enough, my bad... Since using the service is better, I'm gonna add some functions in it as well, for addingAllFacets, and for setting the `defaultFacets` and `allFacets` (these are for the cases where the facets settings is not done in the `config.ts` file but through the administration platform json config, so it cannot be fed to the provider).